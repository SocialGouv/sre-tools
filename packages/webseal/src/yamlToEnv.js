import { load } from "js-yaml";

export const toEnv = (string) => {
  try {
    const obj = load(string);
    if (obj != null && obj.constructor.name === "Object") {
      return Object.entries(obj)
        .map(([key, value]) => `${key}="${value}"`)
        .join("\n");
    }
  } catch (e) {
    console.error(e);
    return string;
  }
  return string;
};
