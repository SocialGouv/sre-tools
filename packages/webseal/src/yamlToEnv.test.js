import { toEnv } from "./yamlToEnv";

const tests = [
  {
    title: "valid YAML",
    input: "A: 42\nB: 43",
    expected: true,
  },
  {
    title: "valid YAML quoted",
    input: 'A: "blablabla"',
    expected: true,
  },
  {
    title: "valid YAML multilines",
    input: `A: 42
B: 424
C: |-
  nlabalbalelkfnzemflkfenz
  zekfjbzkmfjbzemfjbzefmze`,
    expected: true,
  },
  {
    title: "invalid YAML",
    input: "some random text",
    expected: false,
  },
  {
    title: "invalid YAML",
    input: "a=42\nb=43",
    expected: false,
  },
  {
    title: "multiple inputs",
    input: `
SOME_TOKEN: abcRE6jnuXVj234Y65zfzeUipcdO
SOME_SECRET: |-
  -----BEGIN RSA PRIVATE KEY-----
  MIIJKAIBAAKCAlkznfzefekBG6MkFlzDVu/C8+c/XABs4HEQNsz61wdKTv9olitl
  QcJ6USzSsQNymwjgkWkkGH987z33As6c0VnqXZBeW2RZouctk4P3888cUBmC8AX+
  JvmbEZ3IBic1uXkTY8tyXtvdmzBJClTFMqDBr9bwq975Yosj35lmp+JEFLdOsZT/
  -----END RSA PRIVATE KEY-----
SOME_EMAIL: aaa@chez.com, bbb@chez.com
SOME_DB: postgres://user:pass@host:port/db?sslmode=require
`,
    expected: true,
  },
];

tests.forEach((t) => {
  it(`toEnv : ${t.title}`, () => {
    expect(toEnv(t.input)).toMatchSnapshot();
  });
});
