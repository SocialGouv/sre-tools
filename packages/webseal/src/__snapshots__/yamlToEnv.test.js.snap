// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toEnv : invalid YAML 1`] = `"some random text"`;

exports[`toEnv : invalid YAML 2`] = `
"a=42
b=43"
`;

exports[`toEnv : multiple inputs 1`] = `
"SOME_TOKEN=\\"abcRE6jnuXVj234Y65zfzeUipcdO\\"
SOME_SECRET=\\"-----BEGIN RSA PRIVATE KEY-----
MIIJKAIBAAKCAlkznfzefekBG6MkFlzDVu/C8+c/XABs4HEQNsz61wdKTv9olitl
QcJ6USzSsQNymwjgkWkkGH987z33As6c0VnqXZBeW2RZouctk4P3888cUBmC8AX+
JvmbEZ3IBic1uXkTY8tyXtvdmzBJClTFMqDBr9bwq975Yosj35lmp+JEFLdOsZT/
-----END RSA PRIVATE KEY-----\\"
SOME_EMAIL=\\"aaa@chez.com, bbb@chez.com\\"
SOME_DB=\\"postgres://user:pass@host:port/db?sslmode=require\\""
`;

exports[`toEnv : valid YAML 1`] = `
"A=\\"42\\"
B=\\"43\\""
`;

exports[`toEnv : valid YAML multilines 1`] = `
"A=\\"42\\"
B=\\"424\\"
C=\\"nlabalbalelkfnzemflkfenz
zekfjbzkmfjbzemfjbzefmze\\""
`;

exports[`toEnv : valid YAML quoted 1`] = `"A=\\"blablabla\\""`;
